nautobot/dcim/factory.py

```diff
@@ -698,6 +698,7 @@ class Params:
     default_image = factory.LazyAttribute(
         lambda o: not o.software_version.software_image_files.filter(default_image=True).exists()
     )
+    external_integration = random_instance(ExternalIntegration, allow_null=True)
```

Comment:
I do not know how I missed it, but this needs a `has_external_integration` field as well. This will probably fix the unit tests. Similar implementation here https://github.com/nautobot/nautobot/pull/7011/files#diff-78ab3ed8f5c4400737dcdb55ebf9a4e37abe332e36dffa43ecfa3ac889a200f4L759-L762